saleor/graphql/app/types.py

```diff
@@ -2,18 +2,61 @@
 from graphene_federation import key
 
 from ...app import models
+from ...core.exceptions import PermissionDenied
 from ...core.permissions import AppPermission
 from ...core.tracing import traced_resolver
 from ..core.connection import CountableDjangoObjectType
 from ..core.types import Permission
 from ..core.types.common import Job
 from ..meta.types import ObjectWithMetadata
-from ..utils import format_permissions_for_display
+from ..utils import format_permissions_for_display, get_user_or_app_from_context
 from ..webhook.types import Webhook
+from .dataloaders import AppByIdLoader, AppExtensionByAppIdLoader
 from .enums import AppTypeEnum
 from .resolvers import resolve_access_token
 
 
+class AppExtension(CountableDjangoObjectType):
+    app = graphene.Field("saleor.graphql.app.types.App", required=True)
+    permissions = graphene.List(
+        Permission, description="List of the app extension's permissions."
+    )
+
+    class Meta:
+        description = "Represents app data."
+        interfaces = [graphene.relay.Node]
+        model = models.AppExtension
+        only_fields = [
+            "label",
+            "url",
+            "view",
+            "type",
+            "target",
+        ]
+
+    @staticmethod
+    def resolve_app(root, info):
+        app_id = None
+        app = info.context.app
+        if app and app.id == root.app_id:
+            app_id = root.app_id
+
+        requestor = get_user_or_app_from_context(info.context)
+        if requestor.has_perm(AppPermission.MANAGE_APPS):
+            app_id = root.app_id
```

Comment:
Maybe, if the app tries to resolve themself we should skip permissions checks? 

```suggestion
        if app and app.id == root.app_id:
            app_id = root.app_id
        else: 
            requestor = get_user_or_app_from_context(info.context)
            if requestor.has_perm(AppPermission.MANAGE_APPS):
                app_id = root.app_id
```